roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -24,14 +24,26 @@
   # Hard code this so we will skip the entire file instead of individual tasks (Default isn't Consistent)
   static: False
 
+- name: check the journal device is partition
+  shell: "readlink -f {{ item }} | egrep '/dev/([hsv]d[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p)[0-9]{1,2}$'"
+  with_items: "{{ raw_journal_devices }}"
+  changed_when: false
+  failed_when: false
+  always_run: true
+  register: journal_ispartition_results
+
 - name: check the partition status of the journal devices
-  shell: "parted --script {{ item }} print > /dev/null 2>&1"
-  with_items: "{{ raw_journal_devices|unique }}"
+  shell: "parted --script {{ item.1 }} print > /dev/null 2>&1"
+  with_together:
+    - "{{ journal_ispartition_results.results }}"
+    - "{{ raw_journal_devices|unique }}"
   changed_when: false
   failed_when: false
   always_run: true
   register: journal_partition_status
-  when: raw_multi_journal or dmcrypt_dedicated_journal
+  when:
+    - raw_multi_journal or dmcrypt_dedicated_journal
```

Comment:
I'm not 100% sure but you might want to write this like this instead:
`- (raw_multi_journal or dmcrypt_dedicated_journal)`